src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java

```diff
@@ -18,17 +23,65 @@ public void onPlayerMove(PlayerMoveEvent event) {
         IridiumSkyblock.getInstance().getTeamManager().sendIslandBorder(event.getPlayer());
 
         user.getCurrentIsland().ifPresent(island -> {
-            if (event.getPlayer().getLocation().getY() < LocationUtils.getMinHeight(event.getPlayer().getWorld())) {
-                VoidEnhancementData voidEnhancementData = IridiumSkyblock.getInstance().getEnhancements().voidEnhancement.levels.get(IridiumSkyblock.getInstance().getTeamManager().getTeamEnhancement(island, "void").getLevel());
-                if (voidEnhancementData != null && voidEnhancementData.enabled) {
-                    if (IridiumSkyblock.getInstance().getTeamManager().teleport(event.getPlayer(), island.getHome(), island)) {
-                        event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().voidTeleport
-                                .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
-                        ));
+            if (!(event.getPlayer().getLocation().getY() < LocationUtils.getMinHeight(event.getPlayer().getWorld()))) {
+                return;
+            }
+
+            VoidEnhancementData voidEnhancementData = IridiumSkyblock.getInstance()
+                    .getEnhancements().voidEnhancement.levels
+                    .get(IridiumSkyblock.getInstance().getTeamManager().getTeamEnhancement(island, "void").getLevel());
+
+            if (voidEnhancementData == null || !voidEnhancementData.enabled) {
+                return;
+            }
+
+            if (!IridiumSkyblock.getInstance().getTeamManager().teleport(event.getPlayer(), island.getHome(), island)) {
+                return;
+            }
+
+            event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().voidTeleport
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+
+            HashMap<ItemStack, Integer> lostItemsMap = new HashMap<>();
+            if (voidEnhancementData.itemLossChance > 0) {
```

Comment:
We should use early return pattern
```suggestion
            if (voidEnhancementData.itemLossChance <= 0) return
```